### PR TITLE
Improve restore command

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | addition of an interaction on the store restoration command, allowing you to choose the backup if no option is provided. Improved exception handling on all command.
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | test restore command

### CODE REVIEW

File: classes/Commands/CreateBackupCommand.php [hunk 1]
```diff
@@ -64,9 +63,8 @@ protected function execute(InputInterface $input, OutputInterface $output): ?int
 
             return $exitCode;
         } catch (Exception $e) {
-            $this->logger->error('An error occurred during the backup process: ' . $e->getMessage());
-
-            return ExitCode::FAIL;
+            $this->logger->error('An error occurred during the  backup creation process');
```
Comment (Quetzacoalt91) [inline classes/Commands/CreateBackupCommand.php]:
```suggestion
            $this->logger->error('An error occurred during the backup creation process');
```

One redundant space is found

File: classes/Commands/ListBackupCommand.php [hunk 2]
```diff
@@ -77,49 +74,32 @@ protected function execute(InputInterface $input, OutputInterface $output): ?int
 
             return ExitCode::SUCCESS;
         } catch (Exception $e) {
-            $this->logger->error('An error occurred during the backup process: ' . $e->getMessage());
-
-            return ExitCode::FAIL;
+            $this->logger->error('An error occurred during the  backup listing process');
```
Comment (Quetzacoalt91) [inline classes/Commands/ListBackupCommand.php]:
```suggestion
            $this->logger->error('An error occurred during the backup listing process');
```

File: classes/Commands/AbstractBackupCommand.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            $timestamp = $dateTime->getTimestamp();
+
+            return
+                [
+                    'timestamp' => $timestamp,
+                    'datetime' => $this->getFormattedDatetime($timestamp),
+                    'version' => $version,
+                    'filename' => $backupName,
+                ];
+        }
```
Comment (Quetzacoalt91) [inline classes/Commands/AbstractBackupCommand.php]: I think we can even go further, this method will be useful on the web interface as well on the backup selection.

image
image


Can you bring this method close to the existing BackupFinder and BackupManager, and let's take the opportunity to add the related tests?

File: classes/Commands/RestoreCommand.php [hunk 4]
```diff
@@ -80,9 +84,59 @@ protected function execute(InputInterface $input, OutputInterface $output): ?int
...
         }
+
+        $formattedBackups = array_map(function ($backupName) {
+            return $this->parseBackupMetadata($backupName);
+        }, $backups);
+
+        // Most recent first
+        usort($formattedBackups, function ($a, $b) {
+            return $b['timestamp'] <=> $a['timestamp'];
+        });
```
Comment (Quetzacoalt91) [inline classes/Commands/RestoreCommand.php]: I think this part will be useful for the web as well, if you can extract this too.

File: classes/Commands/CreateBackupCommand.php (hunk 1 above)
Comment (M0rgan01) [reply classes/Commands/CreateBackupCommand.php]: outdated

File: classes/Commands/ListBackupCommand.php (hunk 2 above)
Comment (M0rgan01) [reply classes/Commands/ListBackupCommand.php]: outdated

Comment (Quetzacoalt91) [reply classes/Commands/ListBackupCommand.php]: Faster than the review :rocket:

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @M0rgan01 

Thank you for your PR, I tested it and it seems to works as you can see : 

image


Tested from : 
1.7.7.8 to 8.2 
8.0.4 to 8.2 
8.2 to 9.0.0
8.0.4 to 9.0.0

Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Pull request merged**